container_files/demos/Convolutions using SQL and Javascript.ipynb

```diff
@@ -0,0 +1,386 @@
+    "\n",
+    "                              newPixel += inList[y * w + x] * weights[indexW];\n",
+    "                              indexW ++;\n",
+    "                            }\n",
+    "                            \n",
+    "                        new_value = newPixel * alpha;\n",
+    "                        outList[i * w + j] = new_value;\n",
+    "                    }\n",
+    "                return outList;\n",
+    "            } // End of convolution\n",
+    "            \n",
+    "            //Assuring that the 1d row is in the right order\n",
+    "            function arrangeMatrix(inList) {\n",
+    "            \n",
+    "                var length = inList.length;\n",
+    "                var data = new Array(length);\n",
+    "                for (var i = 0; i < length; i++) {\n",
+    "                    data[parseInt(inList[i][0][0])] = inList[i][1];\n",
+    "                }\n",
+    "                return data\n",
+    "            }\n",
+    "            \n",
+    "            /*************************************\n",
+    "            ********** Using Functions ***********\n",
+    "            **************************************/\n",
+    "            \n",
+    "            var weights = arrangeMatrix(kernel); // filter matrix\n",
+    "            var matrix = arrangeMatrix(row_val); // my picture\n",
+    "            var convolvedMatrix = [];\n",
+    "            \n",
+    "            convolution(matrix, convolvedMatrix, dim, dim, alpha, radius);\n",
+    "\n",
+    "            return convolvedMatrix;',\n",
+    "            'val, alpha_, kernel', \n",
+    "             valueExpr, alpha, kernel\n",
+    "        ) AS *\n",
+    "    \"\"\"\n",
+    "\n",
+    "\n",
+    "    print mldb.put(\"/v1/functions/convolution\", {\n",
+    "        \"type\": \"sql.expression\",\n",
+    "        \"params\": {\n",
+    "            \"expression\": JsConvolutionExpr,\n",
+    "            \"prepared\": True\n",
+    "        }\n",
+    "    })\n",
+    "    \n",
+    "create_convolution()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As seen in the code above, each new pixel in the convolved picture is the weighted sum of the the pixel and its neighboring pixels where the weights are the values in the kernel matrix.  \n",
+    "\n",
+    "Doing convolutions with custom function type called [SQL Expression Function](../../../../doc/#builtin/functions/SqlExpressionFunction.md.html) and [jseval](../../../../doc/#builtin/sql/ValueExpression.md.html) for inline definition of functions using Javascript allows us to process large amounts of data using the optimizations inherent to the MLDB platform. Convolutions are typically very time consuming operations with \n",
+    "$O(n.r^2)$ complexity in this case where n is the number of features and r is the radius (i.e. neighboring pixels).\n",
+    "\n",
+    "There were two steps to creating the function above:\n",
```

Comment:
je bougerais ça avant le bloc de code. below there will be these two things. ça va orienter le lecteur